src/createServer.js

```diff
@@ -1,3 +1,68 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { convertToCase } = require('./convertToCase');
+
+function createServer() {
+  const server = http.createServer((req, res) => {
+    const url = new URL(req.url, `http://${req.headers.host}`);
+    const originalText = url.pathname.slice(1);
+    const targetCase = url.searchParams.get('toCase');
+    const errors = [];
+
+    res.setHeader('Content-Type', 'application/json');
+
+    if (!originalText) {
+      errors.push({
+        message:
+          // eslint-disable-next-line max-len
+          'Text to convert is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    }
+
+    if (!targetCase) {
+      errors.push({
+        message:
+          // eslint-disable-next-line max-len
+          '"toCase" query param is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    } else if (
+      !['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'].includes(targetCase)
+    ) {
+      errors.push({
+        message:
+          // eslint-disable-next-line max-len
+          'This case is not supported. Available cases: SNAKE, KEBAB, CAMEL, PASCAL, UPPER.',
+      });
+    }
+
+    if (errors.length) {
+      res.statusCode = 400;
+      res.statusMessage = 'Bad request';
+      res.end(JSON.stringify({ errors }));
+
+      return;
+    }
+
+    const { originalCase, convertedText } = convertToCase(
+      originalText,
+      targetCase,
+    );
+
+    res.statusCode = 200;
+    res.statusMessage = 'OK';
+
+    res.end(
+      JSON.stringify({
+        originalCase,
+        targetCase,
+        originalText,
+        convertedText,
+      }),
+    );
+  });
```

Comment:
This function is doing too many things: it's creating a server, parsing the URL, validating the input, converting the case, and sending the response. Consider breaking it down into smaller, more focused functions to improve readability and maintainability. This is a common principle of clean code - functions should do one thing and do it well.